docs/en/changelog/migration_2_5.rst

```diff
@@ -0,0 +1,690 @@
+
+.. code-block:: php
+
+    <?php
+
+    /**
+     * @Table(name="product", indexes={@Index(columns={"description"},flags={"fulltext"})})
+     */
+    class Product
+    {
+        private $description;
+    }
+
+This feature was contributed by `Adrian Olek <https://github.com/adrianolek>`_.
+
+- `Pull Request <https://github.com/doctrine/doctrine2/pull/973>`_
+
+SQLFilter API: Check if parameter isset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can now check in your SQLFilter if a parameter was set. This allows
+to more easily control which features of a filter to enable or disable.
+
+Extending on the locale example of the documentation:
+
+.. code-block:: php
+
+    <?php
+    class MyLocaleFilter extends SQLFilter
+    {
+        public function addFilterConstraint(ClassMetadata $targetEntity, $targetTableAlias)
+        {
+            if (!$targetEntity->reflClass->implementsInterface('LocaleAware')) {
+                return "";
+            }
+
+            if (!$this->hasParameter('locale')) {
+                return "";
+            }
+
+            return $targetTableAlias.'.locale = ' . $this->getParameter('locale');
+        }
+    }
+
+This feature was contributed by `Miroslav Demovic <https://github.com/mdemo>`_
+
+- `Pull Request <https://github.com/doctrine/doctrine2/pull/963>`_
+
+
+EXTRA_LAZY Improvements
+~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Efficient query when using EXTRA_LAZY and containsKey
+
+    When calling ``Collection::containsKey($key)`` on one-to-many and many-to-many
+    collections using ``indexBy`` and ``EXTRA_LAZY`` a query is now executed to check
+    for the existance for the item. Prevoiusly this operation was performed in memory
+    by loading all entities of the collection.
+
+    .. code-block:: php
+
+        <?php
+
+        class User
+        {
+            /** @OneToMany(targetEntity="Group", indexBy="id") */
+            private $groups;
+        }
+
+        if ($user->getGroups()->containsKey($groupId)) {
+            echo "User is in group $groupId\n";
+        }
+
+    This feature was contributed by `Asmir Mustafic <https://github.com/goetas>`_
+
+    - `Pull Request <https://github.com/doctrine/doctrine2/pull/937>`_
+
+2. Add EXTRA_LAZY Support for get() for owning and inverse many-to-many 
+
+   This was contributed by `Sander Marechal <https://github.com/sandermarechal>`_.
+
+Improve efficiency of One-To-Many EAGER
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When marking a one-to-many association with ``fetch="EAGER"`` it will now
+execute one query less than before and work correctly in combination with
+``indexBy``.
+
+Full support for EntityManagerInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Most of the locations where previously only the ``Doctrine\ORM\EntityManager``
+was allowed are now changed to accept the ``EntityManagerInterface`` that was
+introduced in 2.4. This allows you to more easily use the decorator pattern
+to extend the EntityManager if you need.
```

Comment:
I would be careful when saying this, because we still have lots of places where we pass ``$this`` to other objects, which is not compatible with decoration (the other objects receives the inner manager). And we cannot fix this in the current architecture